source/ui/dataman/AMScanThumbnailGridView.h

```diff
@@ -0,0 +1,209 @@
+#ifndef AMTHUMBNAILGRIDVIEW_H
+#define AMTHUMBNAILGRIDVIEW_H
+
+#include <QAbstractItemView>
+#include <QScrollBar>
+#include <QMouseEvent>
+#include <QRubberBand>
+
+/**
+ * View class which displays a list of Scan Thumbnails in a flow grid style
+ */
+class AMScanThumbnailGridGeometryManager;
+class AMScanThumbnailGridInputManager;
+class AMScanThumbnailGridView : public QAbstractItemView
+{
+	Q_OBJECT
+public:
+	/**
+	 * Creates a new instance of an AMScanThumbnailGridView which has no attached
+	 * model
+	 * @param parent::QObject ~ The parent QObject
+	 */
+	explicit AMScanThumbnailGridView(QWidget *parent = 0);
+
+	/**
+	  * Cleans up resources owned by the AMScanThumbnailGridView
+	  */
+	~AMScanThumbnailGridView();
```

Comment:
Virtual destructor missing here.  Could you please double check to make sure all the destructors are virtual?